index.js

```diff
@@ -1,21 +1,41 @@
 'use strict';
 
 const config = require('./config');
-const amqp = require('amqp-wrapper')(config.amqp);
-const knex = require('knex')(config.db.knex);
 const debug = require('debug')('amqp-log-to-pg');
 
-knex.schema.createTableIfNotExists(config.db.tableName, function (table) {
-  table.increments();
-  table.jsonb('data');
-  table.timestamps(true, true);
-}).then(function () {
-  amqp.connect().then(function () {
-    amqp.consume(onMessage);
-  }).catch(console.error);
-});
+let hasMeta;
+function migrate (knex) {
+  return knex.schema.createTableIfNotExists(config.db.tableName, function (table) {
+    table.increments();
+    table.jsonb('meta');
+    table.jsonb('data');
+    table.timestamps(true, true);
+  }).then(() => {
+    return knex.schema.hasColumn(config.db.tableName, 'meta');
+  });
+}
+
+function main () {
+  const knex = require('knex')(config.db.knex);
+  migrate(knex).then(function (result) {
+    const amqp = require('amqp-wrapper')(config.amqp);
+    hasMeta = result;
+    amqp.connect().then(function () {
+      amqp.consume(onMessage);
+    }).catch(console.error);
+  });
+
+  function onMessage (message, cb) {
```

Comment:
not a biggie but you could get around the need for a global `hasMeta` value by changing this to return a function
```js
amqp.consume(onMessage(hasMeta));

function onMessage (hasMeta) {
  return (message, cb) => {
    // ...
  };
}

// or...
const onMessage = hasMeta => (message, cb) => {
  // ...
};
```